### ISSUE TITLE
Stub matchers in yaml contract are not converted properly to ContractDsl

### ISSUE THREAD

**Issue by mzielinski** [spring-cloud/spring-cloud-contract#1778]
**Describe the bug**

Query parameters defined in YAML format are not converted propertly to ContractDSL when matchers are added

**yaml contract**
```
name: Verify all kind of matchers
request:
  method: POST
  url: /check-matchers/1
  headers:
    Content-Type: application/json
  cookies:
    foo: 2
    bar: 3
  queryParameters:
    limit: 10
    offset: 20
    filter: 'email'
    sort: name
    search: 55
    age: 99
    name: John.Doe
    email: 'bob@email.com'
  matchers:
    url:
      regex: /check-matchers/[0-9]
    queryParameters:
      - key: limit
        type: equal_to
        value: 20
      - key: offset
        type: containing
        value: 20
      - key: sort
        type: equal_to
        value: name
      - key: search
        type: not_matching
        value: '^[0-9]{2}$'
      - key: age
        type: not_matching
        value: '^\\w*$'
      - key: name
        type: matching
        value: 'John.*'
      - key: hello
        type: absent
    cookies:
      - key: foo
        regex: '[0-9]'
      - key: bar
        command: 'equals($it)'
    headers:
      - key: Content-Type
        regex: "application/json.*"
response:
  status: 200
  cookies:
    foo: 1
    bar: 2
  matchers:
    headers:
      - key: Content-Type
        regex: "application/json.*"
    cookies:
      - key: foo
        regex: '[0-9]'
      - key: bar
        command: 'equals($it)'
  headers:
    Content-Type: application/json
```

In memory we see that age, search, name are defined with regexp
![image](https://user-images.githubusercontent.com/4195318/166149780-bab1666a-58f3-404a-bf67-e71e748e27cb.png)

But finally, converted contract does not have matchers on stub side
![image](https://user-images.githubusercontent.com/4195318/166149818-bc771a17-ff1d-4b7f-b6d8-af5c6b52dae8.png)

**Comment by mzielinski** [comment]
The issue when `url` instead of `urlPath` is used. There is already test `org.springframework.cloud.contract.verifier.converter.YamlContractConverterSpec#should convert YAML with REST matchers to DSL` which proofs that `urlPath` works correct.

**Comment by mzielinski** [comment]
@spencergibb Fix is already done and merged → https://github.com/spring-cloud/spring-cloud-contract/commit/1304ccab0b681e126d56c836c631467c7b20a1bf
In which version of SSC I can expect it?

**Comment by marcingrzejszczak** [comment]
![image](https://user-images.githubusercontent.com/3297437/167424881-3174d362-9d06-4f8e-b6f0-5ad44a941e19.png)

This means that it will be released in the next 3.1.x and 4.0.0.

**Comment by marcingrzejszczak** [comment]
Fixed via https://github.com/spring-cloud/spring-cloud-contract/commit/1304ccab0b681e126d56c836c631467c7b20a1bf
